controllers/etcd_custodian_controller.go

```diff
@@ -108,6 +122,37 @@ func (ec *EtcdCustodian) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, err
 	}
 
+	conLength := len(etcd.Status.Conditions)
+	if conLength > 0 && etcd.Status.Conditions[conLength-1].Reason == "QuorumLost" && etcd.Spec.Replicas > 1 {
```

Comment:
I don't think we should consider any earlier conditions from the list. It might be very disastrous

Reply:
I don't mean considering any other condition. What I meant is that the `Ready` condition that gives `Reason == "QuorumLost"` might not always be at `conLength-1` and it might be good if possible to parse through all the conditions and make sure

Reply:
Suppose, there is a quorum loss. Action is taken on it and then there is some other condition appeared. If we are parsing the whole list every time then we may pick old quorum loss case  which is already taken care of. Is it not?

Reply:
The list would have the same conditions. The reasons would change depending on the state of the cluster. 
When we recover from quorum loss for instance, the condition would be updated and hence picking up an old quorum loss case will not happen imo